frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -589,20 +603,34 @@ common_settings.document_metadata_location = {
         genMetadataFolderMenuItem("doc"),
         genMetadataFolderMenuItem("dir"),
         genMetadataFolderMenuItem("hash"),
-        {
-            text = _("Enable hash-based metadata scanning"),
+        { -- hash-based metadata count / TextViewer
+            text = (function()  -- seems like this is running whenever the root menu is opened, not just sdr storage?
+                local hash_file_infos = DocSettings.getHashDirSdrInfos()
+                local text = T(_("Show %1 documents with hash-based metadata"), #hash_file_infos)
+                if #hash_file_infos == 0 then
+                    return T(_("No documents with hash-based metadata"), #hash_file_infos)
+                end
+                if G_reader_settings:readSetting("document_metadata_folder") ~= "hash" then
+                    return "⚠ "..text  -- I'm not sure how to instantly update this after changing to another storage method radio bullet
+                end
+                return text
+            end)(),
+            keep_menu_open = true,
             enabled_func = function()
-                return G_reader_settings:readSetting("document_metadata_folder") ~= "hash"
-            end,
-            checked_func = function()
-                return G_reader_settings:isTrue("document_metadata_hash_enabled")
+                return #DocSettings.getHashDirSdrInfos() ~= 0
             end,
             callback = function()
-                G_reader_settings:flipNilOrFalse("document_metadata_hash_enabled")
-                if G_reader_settings:isTrue("document_metadata_hash_enabled") then
-                    local doc_save = G_reader_settings:readSetting("save_document")
-                    UIManager:show(InfoMessage:new{ text = string.format(hash_filemod_warn, doc_save)})
+                local hash_file_infos = DocSettings.getHashDirSdrInfos()
+                local book_info_list = ""
+                for i, file_info in ipairs(hash_file_infos) do
+                    book_info_list = book_info_list.."\n"..i..". "..file_info
                 end
+                UIManager:show(require("ui/widget/textviewer"):new{
+                    title = #hash_file_infos.." documents with hash-based metadata",
```

Comment:
`T(_("%1 documents with hash-based metadata"), #hash_file_infos)`

Reply:
I still propose to not scan on every menu show/update.